app/javascript/helpers/Axios.jsx

```diff
@@ -31,4 +62,11 @@ const axiosInstance = axios.create(
   },
 );
 
+// Intercepts requests, filters out and forward search params to the API.
+axiosInstance.interceptors.request.use((_config) => {
+  const params = new URLSearchParams(window.location.search);
+  mergeConfigParams(filterParams(params), _config.params);
+  return { ..._config, ...{ params } };
+}, (error) => Promise.reject(error));
+
```

Comment:
This axios request interceptor (middleware) will run automatically for each request and be responsible for:
1. Extracting the current list of search params through the `URLSearchParams` API [MORE DETAILS](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams).
2. Filter out the current list of search params `filterParams(params)`.
3. Merge the filtered params with the config params `mergeConfigParams(filterParams(params), _config.params);`
4. Return a new axios configuration that has the new merged params   `return { ..._config, ...{ params } };` according to the Interceptors API specs [MORE DETAILS](https://axios-http.com/docs/interceptors).
-> This interceptor will then extract, filter search params and merge them with config params and send them on each request to the API.
The API will then can extract those params when needed to do searching, sorting or pagination.